File: JAVA/14.Number_System/p9.java
import java.util.*;

class Demo{

	public static void main(String[] args){
	
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter a number:");
		int num=sc.nextInt();
		int temp=num,dig=0,sum=0;
		
		while(temp>0){
		
			dig=temp%10;
			sum=(sum*10)+dig;
			temp/=10;
		}
		System.out.println("reverse :"+sum);
	}
}

File: Assignments/Assignment_27/p9.c
#include<stdio.h>

int isHarshad(int num){

        int temp=num,sum=0;
        while(temp>0){

                sum+=temp%10;
                temp/=10;
        }if(num%sum==0){

                return 1;
        }else{return 0;}
}

void main(){

        int num;
        printf("Enter a number:\n");
        scanf("%d",&num);
        if(isHarshad(num)){

                printf("%d is a  harshad number\n",num);
        }else{printf("%d is not a harshad number\n",num);}
}
